R/http.R

```diff
@@ -61,14 +61,38 @@ TWIT_method <- function(method, token, api,
 
 #' Pagination
 #' 
+#' @description 
+#' `r lifecycle::badge("experimental")`
+#' These are internal functions used for pagination inside of rtweet.
+#' 
 #' @keywords internal
-#' @param n Maximum number of results to return.
+#' @param token Expert use only. Use this to override authentication for
+#'   a single API call. In most cases you are better off changing the
+#'   default for all calls. See [auth_as()] for details.
+#' @param n Desired number of results to return. Results are downloaded
+#'   in pages when `n` is large; the default value will download a single
+#'   page. Set `n = Inf` to download as many results as possible.
+#'   
+#'   The Twitter API rate limits the number of requests you can perform
+#'   in each 15 minute period. The easiest way to download more than that is 
+#'   to use `retryonratelimit = TRUE`.
+#'   
+#'   You are not guaranteed to get exactly `n` results back. You will get
+#'   fewer results when tweets have been deleted or if you hit a rate limit. 
+#'   You will get more results if you ask for a number of tweets that's not
+#'   a multiple of page size, e.g. if you request `n = 150` and the page
+#'   size is 200, you'll get 200 results back.
 #' @param get_id A single argument function that returns a vector of ids given 
 #'   the JSON response. The defaults are chosen to cover the most common cases,
 #'   but you'll need to double check whenever implementing pagination for
 #'   a new endpoint.
-#' @param max_id String giving id of most recent tweet to return. 
-#'   Can be used for manual pagination.
+#' @param max_id Provide an upper bound for the returned results; all results
+#'   will have an ID less than or equal to `max_id`. If omitted, results will 
+#'   start from the most recent available. Supplying `max_id` allows you to 
+#'   manually paginate the results.
+#' @param since_id Provides a lower-bound for the results returned; all results
+#'   will have an ID greater than or equal to `since_id`. This can generally
+#'   be omitted.
```

Comment:
Should it be a numeric ID or a character? Not sure I understand well what this is but each message or status has an ID this ID increases as more content is published so if one wants content later than X message can use this?

Reply:
Yes, that's right. But once #510 is done, the user shouldn't need to worry about the details. e.g. the code for retrieving more tweets from a search would be something like this:

```R
tw <- search_tweets("#rstats", n = 10000)
tw_more <- search_tweets("#rstats", max_id = max_id(tw))
```

And that makes me realise why you might need `since_id()` — this code would get all the tweets since your last request:

```R
tw_more <- search_tweets("#rstats", max_id = since_id(tw))
```

I'll make all this work in upcoming PRs.